pkg/controllers/provisioningv2/rke2/managesystemagent/managesystemagentplan.go

```diff
@@ -56,3 +78,77 @@ func (h *handler) OnChangeInstallSUC(cluster *rancherv1.Cluster, status rancherv
 		mcc,
 	}, status, nil
 }
+
+// syncSystemUpgradeControllerStatus queries the managed system-upgrade-controller chart and determines if it is properly configured for a given
+// version of Kubernetes. It applies a condition onto the control-plane object to be used by the planner when handling Kubernetes upgrades.
+func (h *handler) syncSystemUpgradeControllerStatus(obj *rkev1.RKEControlPlane, status rkev1.RKEControlPlaneStatus) (rkev1.RKEControlPlaneStatus, error) {
+	sucBundle, sucBundleRetrievalError := h.bundles.Get(fleet.ClustersDefaultNamespace, name.SafeConcatName("mcc", obj.Name, "managed", "system-upgrade-controller"), metav1.GetOptions{})
+	if sucBundleRetrievalError != nil && !errors.IsNotFound(sucBundleRetrievalError) {
+		return status, sucBundleRetrievalError
+	}
+
+	rke2.SystemUpgradeControllerReady.False(&status)
+	// if we couldn't find the bundle then we know it's not ready
+	if sucBundleRetrievalError != nil {
+		return status, nil
+	}
```

Comment:
Generally I would invert these conditions, like so
```suggestion
	sucBundle, err := h.bundles.Get(fleet.ClustersDefaultNamespace, name.SafeConcatName("mcc", obj.Name, "managed", "system-upgrade-controller"), metav1.GetOptions{})
	if errors.IsNotFound(err) {
		// if we couldn't find the bundle then we know it's not ready
		rke2.SystemUpgradeControllerReady.False(&status)
		return status, nil
	}

	if err != nil {
		return status, err
	}
```
Also, we typically set conditions to Unknown while they are in progress, which is sort of opposite from what the current upstream specification is, but it is still established convention.